### PR TITLE
Update web tool for better test experience

### PR DESCRIPTION
**Pull request by WeikangLee**

### CODE REVIEW

File: src/Components/Basic/ImageBlock.tsx [hunk 1]
```diff
@@ -112,8 +112,8 @@ export class ImageBlock extends React.Component<IProps, IState> {
         }
 
         let source = this.props.mode === 'avatar' ?
-            require('../../Assets/Images/Placeholders/avatar_default.png') :
-            require('../../Assets/Images/Placeholders/image_default.png');
+            require('../../Assets/Images/Placeholders/avatar_default@3X.png') :
```
Comment (faceach) [inline src/Components/Basic/ImageBlock.tsx]: Rename resource file name, remove "@3X".

File: src/Models/Abstract/AbstractModel.ts [hunk 2]
```diff
@@ -17,12 +18,39 @@ export abstract class AbstractModel extends TreeNode<AbstractModel> {
...
+
+            if (this.type === undefined) {
+                this.path = 'undefined type >>';
+            } else {
+                this.path = this.type + ' >> ';
+            }
+
+        } else {
+            if (this.type === undefined) {
+                this.path = parent.path + 'undefined type >>';
```
Comment (faceach) [inline src/Models/Abstract/AbstractModel.ts]: I'd prefer to use an array to describe the path, e.g. ["Container", "ColumnSet", "Column", "undefined"]. And format to "Container >> ColumnSet >> Column >> undefined" on outputing

File: src/Components/Basic/ImageBlock.tsx (hunk 1 above)
Comment (DongyuZhao) [inline src/Components/Basic/ImageBlock.tsx]: Please make sure this change will not break mobile render

File: src/Components/Basic/ImageBlock.tsx [hunk 3]
```diff
@@ -112,8 +112,8 @@ export class ImageBlock extends React.Component<IProps, IState> {
...
+            require('../../Assets/Images/Placeholders/image_default@3X.png');
```
Comment (DongyuZhao) [inline src/Components/Basic/ImageBlock.tsx]: Please make sure this change will not break mobile render

File: src/Models/Abstract/AbstractModel.ts [hunk 4]
```diff
@@ -17,12 +18,39 @@ export abstract class AbstractModel extends TreeNode<AbstractModel> {
         this.type = json.type || this.type;
 
         this.schemaCheckResult = this.shallowCheckSchema(json);
+        // this.schemaCheckResult = this.deepCheckSchema();
```
Comment (DongyuZhao) [inline src/Models/Abstract/AbstractModel.ts]: Remove

File: src/Models/Abstract/AbstractModel.ts [hunk 5]
```diff
@@ -17,12 +18,39 @@ export abstract class AbstractModel extends TreeNode<AbstractModel> {
...
+        }
+
+        let temp = undefined;
+        if (parent !== undefined) {
+            temp = parent.type;
+        }
+        console.log(``);
+        console.log(`the parent of ${this.type} is ${temp}`);
+        console.log(`${this.path}`);
+        console.log(``);
```
Comment (faceach) [inline src/Models/Abstract/AbstractModel.ts]: remove

File: src/Models/Abstract/AbstractModel.ts (hunk 4 above)
Comment (faceach) [inline src/Models/Abstract/AbstractModel.ts]: remove

File: src/Models/Abstract/AbstractModel.ts [hunk 6]
```diff
@@ -83,6 +111,10 @@ export abstract class AbstractModel extends TreeNode<AbstractModel> {
         return this.schemaCheckResult.isValid;
     }
 
+    public get getSchemaCheckResult() {
```
Comment (DongyuZhao) [inline src/Models/Abstract/AbstractModel.ts]: Name convention. If you use get accessor, please treat this as a variable.

Comment (DongyuZhao) [inline src/Models/Abstract/AbstractModel.ts]: Why add this method? Just access this.schemaCheckResult could be OK.

File: src/Models/Abstract/InputModel.ts [hunk 7]
```diff
@@ -32,7 +32,7 @@ export abstract class InputModel extends ContentModel {
     }
 
     public storeListener = (field: FormField) => {
-        console.log(field);
+        // console.log(field);
```
Comment (DongyuZhao) [inline src/Models/Abstract/InputModel.ts]: Remove

File: src/Schemas/SchemaValidator.ts [hunk 8]
```diff
@@ -97,7 +97,8 @@ export class SchemaValidator {
 
             return result;
         }
-        return new SchemaResult(true, new SchemaMessage('Warning', `No Schema available for ${JSON.stringify(json)}`));
+        // return new SchemaResult(true, new SchemaMessage('Warning', `No Schema available for ${JSON.stringify(json)}`));
```
Comment (DongyuZhao) [inline src/Schemas/SchemaValidator.ts]: Remove

File: src/Models/Abstract/InputModel.ts (hunk 7 above)
Comment (faceach) [inline src/Models/Abstract/InputModel.ts]: Remove all unnecessary comment out code.

File: src/Schemas/SchemaValidator.ts (hunk 8 above)
Comment (faceach) [inline src/Schemas/SchemaValidator.ts]: remove

File: src/Schemas/SchemaValidator.ts [hunk 9]
```diff
@@ -97,7 +97,8 @@ export class SchemaValidator {
...
+        return new SchemaResult(true, new SchemaMessage('Warning', `No Schema available for the type, check type!`));
```
Comment (faceach) [inline src/Schemas/SchemaValidator.ts]: which type?

File: src/DataValidator/JsonValidator.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import { CardContext } from '../Contexts/CardContext';
+import { CardModel } from '../Models/Cards/Card';
+
+export class JsonValidator {
+
+    public static isSchemaValid(Json: any) {
+        let context = CardContext.createInstance();
+        let card = new CardModel(Json, undefined, context);
```
Comment (faceach) [inline src/DataValidator/JsonValidator.ts]: Use `const` for these variables

File: src/DataValidator/JsonValidator.ts [hunk 11]
```diff
@@ -0,0 +1,51 @@
+import { CardContext } from '../Contexts/CardContext';
+import { CardModel } from '../Models/Cards/Card';
+
+export class JsonValidator {
+
+    public static isSchemaValid(Json: any) {
```
Comment (faceach) [inline src/DataValidator/JsonValidator.ts]: `camelCase` for parameter variable

File: src/DataValidator/JsonValidator.ts [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    public static getDescendsAndSelf(Json: any) {
+        let context = CardContext.createInstance();
+        let card = new CardModel(Json, undefined, context);
+        return card.descendsAndSelf;
+    }
+
+    public static getSchemaCheckMessage(Json: any) {
+        let checkResult = this.getSchemaCheckResult(Json);
+        let output_message: string[] = [];
+
```
Comment (faceach) [inline src/DataValidator/JsonValidator.ts]: Keep variable name consistant, use `camelCase`

File: src/Models/Abstract/AbstractModel.ts [hunk 13]
```diff
@@ -23,6 +24,24 @@ export abstract class AbstractModel extends TreeNode<AbstractModel> {
         }
 
         this.outputSchemaMessage();
+
+        if (parent === undefined) {
+
```
Comment (faceach) [inline src/Models/Abstract/AbstractModel.ts]: rm blank line

**Pull request merged**